columnar/src/writer/column_operation.rs

```diff
@@ -0,0 +1,311 @@
+use crate::dictionary::UnorderedId;
+use crate::utils::{place_bits, pop_first_byte, select_bits};
+use crate::value::NumericalValue;
+use crate::{DocId, NumericalType};
+
+/// When we build a columnar dataframe, we first just group
+/// all mutations per column, and append them in append-only object.
+///
+/// We represents all of these operations as `ColumnOperation`.
+#[derive(Eq, PartialEq, Debug, Clone, Copy)]
+pub(crate) enum ColumnOperation<T> {
+    NewDoc(DocId),
+    Value(T),
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+struct ColumnOperationHeader {
+    typ_code: u8,
```

Comment:
```suggestion
struct ColumnOperationMetaData {
   operation_type: ColumnOperationType,
```

Header sounds like a one-time thing
```rust
enum ColumnOperationType{
    NewDoc,
    NewValue
} 
````